### ISSUE TITLE
Steam Won't Launch On Pop!_OS 20.10

### ISSUE THREAD

**Issue by JoshuaBillson** [ValveSoftware/steam-for-linux#7847]
#### Your system information

* Steam client version (build number or date): 1623193086
* Distribution (e.g. Ubuntu): Pop!_OS 20.10
* Opted into Steam client beta?: [Yes/No] No
* Have you checked for system updates?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:
The steam client does not launch after updating. If I reinstall Steam, it will again launch a single time until it automatically updates at which point it will again refuse to launch. I am not opted in to the Beta program and I do not have an NVIDIA GPU. Launching from the terminal prints the following log:

Running Steam on pop 20.10 64-bit
STEAM_RUNTIME is enabled automatically
Pins up-to-date!
Steam client's requirements are satisfied
WARNING: Using default/fallback debugger launch
/home/username/.steam/debian-installation/ubuntu12_32/steam -nominidumps -nobreakpad
[2021-06-09 05:03:27] Startup - updater built Jun 8 2021 22:23:36
Looks like steam didn't shutdown cleanly, scheduling immediate update check
[2021-06-09 05:03:27] Loading cached metrics from disk (/home/jmbillson/.steam/debian-installation/package/steam_client_metrics.bin)
[2021-06-09 05:03:27] Using the following download hosts for Public, Realm steamglobal
[2021-06-09 05:03:27] 1. https://cdn.akamai.steamstatic.com, /client/, Realm 'steamglobal', weight was 100, source = 'update_hosts_cached.vdf'
[2021-06-09 05:03:27] 2. https://cdn.cloudflare.steamstatic.com, /client/, Realm 'steamglobal', weight was 100, source = 'update_hosts_cached.vdf'
[2021-06-09 05:03:27] 3. http://media.steampowered.com, /client/, Realm 'steamglobal', weight was 1, source = 'baked in'
[2021-06-09 05:03:27] Checking for update on startup
[2021-06-09 05:03:27] Checking for available updates...
[2021-06-09 05:03:27] Downloading manifest: https://cdn.akamai.steamstatic.com/client/steam_client_ubuntu12
[2021-06-09 05:03:28] Download skipped: /client/steam_client_ubuntu12 version 1623193086, installed version 1623193086, existing pending version 0
[2021-06-09 05:03:28] Nothing to do
[2021-06-09 05:03:28] Verifying installation...
[2021-06-09 05:03:28] Performing checksum verification of executable files
[2021-06-09 05:03:28] Verification complete
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy

**Comment by kisak-valve** [comment]
Hello @JoshuaBillson, can you check if you're seeing #7813? In particular, if you run `steam --reset`, then login without telling Steam to remember the password, does it continue to work the next time you start Steam? (https://github.com/ValveSoftware/steam-for-linux/issues/7813#issuecomment-856967759)

**Comment by dwarfdigdeep** [comment]
I have the same output as JoshuaBillson (ends at the `STEAM_RUNTIME_HEAVY` line w/o errors like in the #7813 report).

`steam --reset` does nothing for me: login never happens (very briefly a steam updater progress bar flashes) and log output is the same.

**Comment by luzfcb** [comment]
@JoshuaBillson Try running Steam with the command:

```
STEAM_RUNTIME=0 steam
```

It shows more useful information.

**Comment by kisak-valve** [comment]
Hello @luzfcb, running Steam with the Steam runtime is unsupported and is unlikely to help.

**Comment by dwarfdigdeep** [comment]
`STEAM_RUNTIME=0 steam` does return additional error information in my case:

> Error: You are missing the following 32-bit libraries, and Steam may not run:
> libXtst.so.6
> libgtk-x11-2.0.so.0
> libpipewire-0.3.so.0

None of which are installed, but I can't test installing them until this evening (they appear to be unavailable in 20.04 which I'm on, but available in 20.10 and 21.04).

**Comment by luzfcb** [comment]
Update: In my case, running `steam --reset` solves the problem of Steam not opening.
Note: I haven't installed `libpipewire-0.3:i386` yet, since there is no i386 package published in Ubuntu repository

**Comment by luzfcb** [comment]
Why Steam does not provides `--help` and show quick documentation about all command-line options?
![image](https://user-images.githubusercontent.com/807599/121383574-eea5a600-c91d-11eb-9646-937cfe871ad1.png)

**Comment by cybik** [comment]
@luzfcb there may be two issues at play:

* libpipewire may be allowed to fail silently (heh. silently.) as using `--reset` goes through just fine
* the `--reset` and subsequent care in not saving the password as a countermeasure to the crash would get past #7813, which I think is tue true issue at hand here

**Comment by ctorgalson** [comment]
Seeing exactly the same thing on Ubuntu 20.04:

 - Steam downloaded an update this morning,
 - Launching Steam via the UI _did_ show a window very briefly, but no longer does even that,
 - Trying to launch Steam via terminal (with or without the `--reset` flag) shows me the same output posted elsewhere here:
 
   <details>
     <summary>Output of <code>steam</code> AND <code>steam --reset</code> on cli</summary>
   
     ```
     ➜ steam
     Running Steam on ubuntu 20.04 64-bit
     STEAM_RUNTIME is enabled automatically
     Pins up-to-date!
     Steam client's requirements are satisfied
     WARNING: Using default/fallback debugger launch
     /home/ctorgalson/.local/share/Steam/ubuntu12_32/steam -nominidumps -nobreakpad
     [2021-06-09 18:11:54] Startup - updater built Jun  8 2021 22:23:36
     Looks like steam didn't shutdown cleanly, scheduling immediate update check
     [2021-06-09 18:11:54] Loading cached metrics from disk (/home/ctorgalson/.local/share/Steam/package/steam_client_metrics.bin)
     [2021-06-09 18:11:54] Using the following download hosts for Public, Realm steamglobal
     [2021-06-09 18:11:54] 1. https://cdn.akamai.steamstatic.com, /client/, Realm 'steamglobal', weight was 100, source = 'update_hosts_cached.vdf'
     [2021-06-09 18:11:54] 2. https://cdn.cloudflare.steamstatic.com, /client/, Realm 'steamglobal', weight was 100, source = 'update_hosts_cached.vdf'
     [2021-06-09 18:11:54] 3. http://media.steampowered.com, /client/, Realm 'steamglobal', weight was 1, source = 'baked in'
     [2021-06-09 18:11:54] Checking for update on startup
     [2021-06-09 18:11:54] Checking for available updates...
     [2021-06-09 18:11:54] Downloading manifest: https://cdn.akamai.steamstatic.com/client/steam_client_ubuntu12
     [2021-06-09 18:11:54] Download skipped: /client/steam_client_ubuntu12 version 1623193086, installed version 1623193086, existing pending version 0
     [2021-06-09 18:11:54] Nothing to do
     [2021-06-09 18:11:54] Verifying installation...
     [2021-06-09 18:11:54] Performing checksum verification of executable files
     [2021-06-09 18:11:55] Verification complete
     STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
     ```
   
   </details>
 - Trying to launch Steam with `STEAM_RUNTIME=0` displays an error about missing 32-bit packages:
 
    <details>
       <summary>Output of <code>STEAM_RUNTIME=0 steam</code> on cli</summary>
    
       ```
       ➜ STEAM_RUNTIME=0 steam
       Running Steam on ubuntu 20.04 64-bit
       STEAM_RUNTIME is disabled by the user
       Error: You are missing the following 32-bit libraries, and Steam may not run:
       libXtst.so.6
       libXrandr.so.2
       libXrender.so.1
       libgobject-2.0.so.0
       libglib-2.0.so.0
       libgio-2.0.so.0
       libgtk-x11-2.0.so.0
       libpipewire-0.3.so.0
       libpulse.so.0
       libgdk_pixbuf-2.0.so.0
       libva.so.2
       libbz2.so.1.0
       libva-drm.so.2
       libva.so.2
       libva-x11.so.2
       libvdpau.so.1
       Can't find 'steam-runtime-check-requirements', continuing anyway
       WARNING: Using default/fallback debugger launch
       /home/ctorgalson/.local/share/Steam/ubuntu12_32/steam -nominidumps -nobreakpad
       [2021-06-09 18:15:07] Startup - updater built Jun  8 2021 22:23:36
       Looks like steam didn't shutdown cleanly, scheduling immediate update check
       [2021-06-09 18:15:07] Loading cached metrics from disk (/home/ctorgalson/.local/share/Steam/package/steam_client_metrics.bin)
       [2021-06-09 18:15:07] Using the following download hosts for Public, Realm steamglobal
       [2021-06-09 18:15:07] 1. https://cdn.akamai.steamstatic.com, /client/, Realm 'steamglobal', weight was 100, source = 'update_hosts_cached.vdf'
       [2021-06-09 18:15:07] 2. https://cdn.cloudflare.steamstatic.com, /client/, Realm 'steamglobal', weight was 100, source = 'update_hosts_cached.vdf'
       [2021-06-09 18:15:07] 3. http://media.steampowered.com, /client/, Realm 'steamglobal', weight was 1, source = 'baked in'
       [2021-06-09 18:15:07] Checking for update on startup
       [2021-06-09 18:15:07] Checking for available updates...
       [2021-06-09 18:15:07] Downloading manifest: https://cdn.akamai.steamstatic.com/client/steam_client_ubuntu12
       [2021-06-09 18:15:08] Download skipped: /client/steam_client_ubuntu12 version 1623193086, installed version 1623193086, existing pending version 0
       [2021-06-09 18:15:08] Nothing to do
       [2021-06-09 18:15:08] Verifying installation...
       [2021-06-09 18:15:08] Performing checksum verification of executable files
       [2021-06-09 18:15:08] Verification complete
       ```
       
    </details>

**Comment by smcv** [comment]
Error messages about missing 32-bit packages when you run with `STEAM_RUNTIME=0` are not relevant here. You are seeing those error messages *because* you are using `STEAM_RUNTIME=0`.

The reason those libraries are missing is because you are specifically telling Steam not to use the copy of those libraries provided by the Steam Runtime. This is not a supported thing to do, and is just distracting you.

**Comment by smcv** [comment]
https://github.com/ValveSoftware/steam-for-linux/issues/7813#issuecomment-857851359 has some general comments about crashing on startup which are also applicable here.

**Comment by ctorgalson** [comment]
> Error messages about missing 32-bit packages when you run with STEAM_RUNTIME=0 are not relevant here. 

Understood. Only added that there because it was suggested upthread as something that might expose more information.

**Comment by luzfcb** [comment]
@smcv I'm not using Steam Beta at all. Since the https://github.com/ValveSoftware/steam-for-linux/issues/7813#issuecomment-857851359 issue explicitly cites Steam Beta, it might be better to also make it explicit if your comment applies to regular Steam or only to the Steam Beta. (I like a [Python Zen](https://www.python.org/dev/peps/pep-0020/#id2) phrase: Explicit is better than implicit). Anyway, thanks for your extremely detailed comment.

**Comment by smcv** [comment]
#7813 was originally about the Steam beta, but people have left comments there about both the beta and the "general availability" (non-beta) version, and I think different people commenting on that issue might be talking about more than one reason to crash.

To make this explicit: everything I said in https://github.com/ValveSoftware/steam-for-linux/issues/7813#issuecomment-857851359 is equally applicable to the beta and the "general availability" (non-beta) version.

**Comment by dwarfdigdeep** [comment]
In my case I can get steam to launch once by removing the `~/.steam` directory. After exiting out, attempting to run Steam again results in the same log output as the original post (no errors or other indication of a fault).

I'm not using the beta client and I have run the `setup.sh --force` command.

**Comment by timshannon** [comment]
Same for me, but that only works once, after which it fails to start again.  `steam --reset` also doesn't work for me.

**Comment by JoshuaBillson** [comment]
> Hello @JoshuaBillson, can you check if you're seeing #7813? In particular, if you run `steam --reset`, then login without telling Steam to remember the password, does it continue to work the next time you start Steam? ([#7813 (comment)](https://github.com/ValveSoftware/steam-for-linux/issues/7813#issuecomment-856967759))

Thanks for the help @kisak-valve, that seems to have solved my problem. I have restarted my PC several times and launched a few games to confirm that the issue does not return and so far everything seems good. The issue seems to be related to the login process; so long as I do not instruct Steam to remember my credentials everything continuous to work.

**Comment by JoshuaBillson** [comment]
> In my case I can get steam to launch once by removing the `~/.steam` directory. After exiting out, attempting to run Steam again results in the same log output as the original post (no errors or other indication of a fault).
> 
> I'm not using the beta client and I have run the `setup.sh --force` command.

This was exactly my problem; steam would launch once and then crash from that point on once I closed the client. Deleting ~/.steam or re-installing steam would let me get back into the client but then the issue would return once I closed it again. The solution proposed by kisak-valve seems to have resolved my issue however. The trick seems to be that you must instruct Steam to not remember your login credentials. To do this, once you get into the client go to Steam > Settings > Account and make sure you check the box which says “Don’t save account credentials on this computer”.

**Comment by smcv** [comment]
As a data point, I have Steam version 1623193086 on an Ubuntu 20.04 test system here and it's working fine, so there must be *something* about @ctorgalson's system that is not the same. I'm using an existing `~/.steam`, I'm using saved account credentials, and I don't have `libpipewire-0.3-0:i386` or `libnm0:i386` installed, so in theory I should have all the risk factors that people are saying could be contributing to Steam not starting, but I still can't reproduce the crash.

I tried creating a new user account with an empty home directory and logging in to Steam in that user account, but that also seems to be working fine.

For people who are affected by Steam crashing on startup: getting information from `steam-runtime-system-info` as described in https://github.com/ValveSoftware/steam-for-linux/issues/7813#issuecomment-857851359 might help Steam developers to figure out why you get that crash and I don't.

@JoshuaBillson and @luzfcb say that they can work around this by using `steam --reset` and then not letting Steam save their credentials. If that workaround is successful for you too, then you are maybe experiencing the same issue as them.

If that workaround is not successful, then you might be seeing the same symptom (a crash) but for a different reason - there are lots of reasons why a program as complicated as Steam can crash! If that's the case, then it is probably better to open a separate issue with full details of your system, what you are experiencing, what workarounds you tried, etc., so that it can be dealt with separately if necessary.

**Comment by dwarfdigdeep** [comment]
@JoshuaBillson works like a charm! I'm not sure why `steam --reset` does nothing for me, but removing `~/.steam` and choosing NOT to save my login appears to resolve the issue (relaunched steam several times and launched a few games).

**Comment by ctorgalson** [comment]
Possible data point:

- `steam --reset` changes nothing here,
- removing `~/.steam` and launching Steam works,
- at this point, it becomes _impossible_ to save the password (I can log in successfully, but if I check the box, I'm prompted to re-login next time anyway).

**Comment by JoshuaBillson** [comment]
> Possible data point:
> 
> * `steam --reset` changes nothing here,
> * removing `~/.steam` and launching Steam works,
> * at this point, it becomes _impossible_ to save the password (I can log in successfully, but if I check the box, I'm prompted to re-login next time anyway).

Did you check the box which says “Don’t save account credentials on this computer” in Steam > Settings > Account? If so, I would imagine this is why it won't remember your password even when you instruct it to. Also, I want to point out that an alternative to removing ~/.steam is to simply uninstall and reinstall the Steam client by running "sudo apt-get remove steam && sudo apt-get install steam".

**Comment by ctorgalson** [comment]
> Did you check the box which says

:man_facepalming: I did.

With that change, I'm in the camp of "everything works great without saving logins".

**Comment by janstuemmel** [comment]
> @JoshuaBillson works like a charm! I'm not sure why `steam --reset` does nothing for me, but removing `~/.steam` and choosing NOT to save my login appears to resolve the issue (relaunched steam several times and launched a few games).

That works for me as well, but when i install/start a game (csgo, it wont start) and then close and restart steam, same problem again...

**Comment by compumike** [comment]
I'm also experiencing this problem. My temporary workaround was to edit `~/.steam/registry.vdf` and change `Registry.HKCU.Software.Valve.Steam.RememberPassword` from `"1"` to `"0"`. After that, I was successfully able to launch Steam (and was asked for my password).

**Comment by wadtech** [comment]
> I'm also experiencing this problem. My temporary workaround was to edit `~/.steam/registry.vdf` and change `Registry.HKCU.Software.Valve.Steam.RememberPassword` from `"1"` to `"0"`. After that, I was successfully able to launch Steam (and was asked for my password).

Neat solution thank you. This worked for me on Ubuntu 21.04 where previous suggestions didn't work.

**Comment by janstuemmel** [comment]
steam updated, works for me now as expected

**Comment by florianstancioiu** [comment]
I have the same problem on Ubuntu 20.04, nothing updated for me - it doesn't work for like 3 days straight.

**Comment by ParoXoN** [comment]
Experiencing the same symptoms under Ubuntu 20.04.

Previously, removing docker interfaces and running `steam --reset` would get it running again, but not today.

Was finally able to get back into steam after editing `~/.steam/registry.vdf` to disable `Registry.HKCU.Software.Valve.Steam.RememberPassword` (thanks @compumike !)

**Comment by kisak-valve** [comment]
The issue in the opening post of this issue report should be fixed by the [2021-07-13](https://store.steampowered.com/oldnews/86266) Steam client update.

Closing as fixed and a duplicate of #7813.
